### PR TITLE
fix: Fixed an error when the loading property of the Button component…

### PR DESCRIPTION
**Pull request by kejianfeng**
… was passed into null

[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

[https://github.com/ant-design/ant-design/issues/36282](https://github.com/ant-design/ant-design/issues/36282)

### 💡 Background and solution

- background:  An error occurs when the loading attribute of the Button component is passed into null
- solution： I hope to change typeof loading === 'object' in the source code to Object.prototype.toString.call(loading)
This will not report an error even if it is null. In this way, the scene component itself that will report an error should be avoided. No matter what parameters are passed in, the error will not cause the page to crash directly.

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     When determining whether the passed loading parameter is an object, change typeof to Object.prototype.toString.call     |
| 🇨🇳 Chinese |      判断传入的loading参数是否为对象时，把 typeof 改为 Object.prototype.toString.call     |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by yoyo837** [comment]
How about using `Optional Chaining Operator`？

File: components/button/button.tsx [hunk 1]
```diff
@@ -190,7 +190,7 @@ const InternalButton: React.ForwardRefRenderFunction<unknown, ButtonProps> = (pr
 
   // =============== Update Loading ===============
   const loadingOrDelay: Loading =
-    typeof loading === 'object' && loading.delay ? loading.delay || true : !!loading;
+    Object.prototype.toString.call(loading) === '[object Object]' && loading.delay ? loading.delay || true : !!loading;
```
Comment (afc163) [inline components/button/button.tsx:193]:
```suggestion
      loading?.delay ?? !!loading;
```

这样应该就行了。

File: components/button/button.tsx [hunk 2]
```diff
@@ -190,7 +190,7 @@ const InternalButton: React.ForwardRefRenderFunction<unknown, ButtonProps> = (pr
 
   // =============== Update Loading ===============
   const loadingOrDelay: Loading =
-    Object.prototype.toString.call(loading) === '[object Object]' && loading.delay ? loading.delay || true : !!loading;
+      loading?.delay ?? !!loading;
```
Comment (afc163) [inline components/button/button.tsx:193]:
```suggestion
    loading?.delay ?? !!loading;
```

Comment (MadCcc) [reply components/button/button.tsx:193]: 断言不如还是判断下类型？我觉得断言都是有隐患的。。

Comment (afc163) [reply components/button/button.tsx:193]:
```suggestion
     typeof loading === 'boolean' ? loading : (loading?.delay || true);
```

File: components/button/button.tsx [hunk 3]
```diff
@@ -190,7 +190,7 @@ const InternalButton: React.ForwardRefRenderFunction<unknown, ButtonProps> = (pr
 
   // =============== Update Loading ===============
   const loadingOrDelay: Loading =
-    typeof loading === 'object' && loading.delay ? loading.delay || true : !!loading;
+      typeof loading === 'boolean' ? loading : loading?.delay;
```
Comment (afc163) [inline components/button/button.tsx:193]:
```suggestion
      typeof loading === 'boolean' ? loading : (loading?.delay || true);
```

**Comment by kejianfeng** [comment]
Using Optional Chaining Operator can solve the problem, but in order to show that the purpose is to judge a real object, it is more reasonable to use Object.prototype.toString.call.


在 2022年6月29日 ***@***.***> 写道：


How about using Optional Chaining Operator？

**Pull request merged**